contracts/voting-verifier/src/query.rs

```diff
@@ -1,28 +1,256 @@
+use axelar_wasm_std::voting::{Poll, PollStatus};
 use connection_router::state::{CrossChainId, Message};
+use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{Deps, StdResult};
 
 use crate::error::ContractError;
-use crate::state::VERIFIED_MESSAGES;
+use crate::state::{self, POLLS, POLL_MESSAGES};
 
-pub fn verification_statuses(
-    deps: Deps,
-    messages: Vec<Message>,
-) -> StdResult<Vec<(CrossChainId, bool)>> {
+#[cw_serde]
+pub enum VerificationStatus {
+    Verified,
+    NotVerified,
+    InProgress, // still in an open poll
+    None,       // not in a poll
+}
```

Comment:
why do we need all these states now?

Reply:
it's used by the execute function to check if the messages requested to verify were already verified or not. Before we just checked `VERIFIED_MESSAGES` storage, messages could be in multiple polls in progress but would end up in a single place in the `VERIFIED_MESSAGES`, but now the calculation is done on the fly and we need to check the poll where the message is being voted on to check it's verification status, so we can't allow the same message to be in multiple polls in progress at the same time. We only allow a message to be part of a poll if its not verified and is not part of a poll in progress